controllers/v1/cloud-services/files.js

```diff
@@ -79,110 +78,162 @@ module.exports = class Files {
+   *  "message": "Url's generated successfully",
+   *  "result": [{
+   *  "filePath": "5e1c28a050452374e1cf9841/e97b5582-471c-4649-8401-3cc4249359bb/cdv_photo_117.jpg",
+   *  "url": "https://storage.googleapis.com/download/storage/v1/b/sl-dev-storage/o/5e1c28a050452374e1cf9841%2Fe97b5582-471c-4649-8401-3cc4249359bb%2Fcdv_photo_117.jpg?generation=1579240054787924&alt=media"
+   * }]
+   */
+
+  /**
+   * Get Downloadable URL from cloud service.
+   * @method
+   * @name getDownloadableUrl
+   * @param  {Request}  req  request body.
+   * @returns {JSON} Response with status and message.
+   */
+
+  async getDownloadableUrl(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let downloadableUrl = await filesHelpers.getDownloadableUrl(
+          req.body.filePaths
+        );
+
+        return resolve(downloadableUrl);
+      } catch (error) {
+        return reject({
+          status:
+            error.status || httpStatusCode["internal_server_error"].status,
+
+          message:
+            error.message || httpStatusCode["internal_server_error"].message,
+
+          errorObject: error,
+        });
+      }
+    });
+  }
+
+  /**
+      * upload the file to the cloud .
       * @method
-      * @name preSignedUrls
+      * @name upload
       * @param  {Request}  req  request body.
-      * @param  {Array}  req.body.fileNames - list of file names
-      * @param  {String}  req.body.bucket - name of the bucket  
-      * @returns {JSON} Response with status and message.
-    */
+      * @returns {JSON} Response with status .
+      * @apiParamExample {json} Response:
+     * {
+        "status": 200
+       }
+     */
 
-   async preSignedUrls(req) {
+  async upload(req) {
     return new Promise(async (resolve, reject) => {
-
-        try {
-
-            let signedUrl =
-            await filesHelpers.preSignedUrls(
-                 req.body.request,
-                 req.body.ref,
-                 req.userDetails ? req.userDetails.userId : ""
+      try {
+        let currentMoment = moment(new Date());
+        let formattedDate = currentMoment.format("DD-MM-YYYY");
+        if (!fs.existsSync(`${ROOT_PATH}/public/assets/${formattedDate}`)) {
+          fs.mkdirSync(`${ROOT_PATH}/public/assets/${formattedDate}`);
+        }
+        let filename = path.basename(req.query.file);
+        let localFilePath = `${ROOT_PATH}/public/assets/${formattedDate}/${filename}`;
+        let uploadStatus
+        // Use the Promise version of fs.writeFile
+        
+        if (req.headers["content-type"] === "application/octet-stream") {
+          const binaryData = Buffer.from(req.body);
+          await fs.promises.writeFile(localFilePath, binaryData);
+          uploadStatus = await filesHelpers.upload(localFilePath, req.query.file);
+
+          if(uploadStatus.status == 200){
+            return resolve({
+              status: httpStatusCode["ok"].status,
+            });
+          }
+        } else {
```

Comment:
@ankitpws Make this condition for mutlipart/form-data only.

Reply:
Done